### PR TITLE
[WIP] Only emit `model` event after ledger closed

### PR DESCRIPTION
**Pull request by sublimator**

### CODE REVIEW

File: src/core/orderbook.js [hunk 1]
```diff
@@ -123,6 +125,13 @@ function OrderBook(remote,
     self.updateFundedAmounts(transaction);
   }
 
+  function ledgerClosed(message) {
+    if (message.ledger_index === self._ledgerLastNotified) {
+      // We had transaction notifications, so notify and recompute
```
Comment (sublimator) [inline src/core/orderbook.js]: We might do something similar with all the `offer_added` events, renaming them to `_offer_added` and buffering these events until ledger close.

@shekenahglory  

How are the offer_added etc events used in ripple-charts (if at all)?

**Comment by sublimator** [comment]
This would only really help if you had lots of transactions for the same book in one ledger.

File: src/core/orderbook.js [hunk 2]
```diff
@@ -38,9 +38,10 @@ function assertValidNumber(number, message) {
  * @param {String} orderbook key
  */
 
-function OrderBook(remote,
-                   currencyGets, issuerGets, currencyPays, issuerPays,
-                   key) {
+/* eslint-disable max-len*/
+// Otherwise it gets confused by the indentation!
```
Comment (clark800) [inline src/core/orderbook.js]: One option that I've been using is to put the `) {` on a line of its own.

**Comment by clark800** [comment]
@sublimator can you add a comment explaining why this was closed?

**Comment by sublimator** [comment]
It was a quick kludge, and I believe Ivan and Mathew are working on fixes
for the perf issue, in the context of actual apps they are working on.

The idea was basically to buffer txn notifications, until the ledger
closed.  Note all the `"type" : "transaction"` messages are for validated
messages, and arrive in quick succession, followed the ledgerClose.

I think Matt ended up going with a more elegant simple throttle. I haven't
really been following along.

**Comment by clark800** [comment]
I think the buffering solution seems more elegant than the throttling solution. I think Ivan may be working on changing to a buffering solution now. I linked him to this pull request.

**Comment by sublimator** [comment]
Whatever works

**Comment by sublimator** [comment]
If we are spending a lot of time on this, we should also update the book requests to request a validated ledger.

**Pull request closed without merging**